tests/test_report_manager.py

```diff
@@ -12,141 +12,174 @@
         ]
-
         yaml_content = """
-            num_configs_per_model: 5
+            num_configs_per_model: """ + str(num_configs_per_model) + """
             client_protocol: grpc
             export_path: /test/export/path
             constraints:
               perf_latency:
                 max: 100
         """
         config = self._evaluate_config(args, yaml_content)
-        self.report_manager = ReportManager(config=config)
+        state_manager = AnalyzerStateManager(config=config)
+        state_manager.load_checkpoint()
+        self.result_manager = ResultManager(config=config,
+                                            state_manager=state_manager)
+        self.report_manager = ReportManager(config=config,
+                                            result_manager=self.result_manager)
+
+    def setUp(self):
+        self.model_config = {
+            "platform": "tensorflow_graphdef",
+            "instance_group": [{
+                "count": 1,
+                "kind": "KIND_GPU"
+            }],
+            "dynamic_batching": {
+                "preferred_batch_size": [4, 8],
+            }
+        }
+
+        self.os_mock = MockOSMethods(mock_paths=[
+            "model_analyzer.reports.report_manager",
+            "model_analyzer.config.input.config_command_analyze"
+        ])
+        self.os_mock.start()
+        self.io_mock = MockIOMethods(
+            mock_paths=["model_analyzer.reports.pdf_report"],
+            read_data=[bytes(">:(".encode("ascii"))])
+        self.io_mock.start()
+        self.matplotlib_mock = MockMatplotlibMethods()
+        self.matplotlib_mock.start()
 
     def test_add_results(self):
-        objective_spec = {'perf_throughput': 10}
-        self.result_comparator = ResultComparator(
-            metric_objectives=objective_spec)
+        self._init_managers("test_model1,test_model2")
+        result_comparator = ResultComparator(
+            metric_objectives={"perf_throughput": 10})
 
-        avg_gpu_metrics = {0: {'gpu_used_memory': 6000, 'gpu_utilization': 60}}
+        avg_gpu_metrics = {0: {"gpu_used_memory": 6000, "gpu_utilization": 60}}
 
         for i in range(10):
             avg_non_gpu_metrics = {
-                'perf_throughput': 100 + 10 * i,
-                'perf_latency': 4000
+                "perf_throughput": 100 + 10 * i,
+                "perf_latency": 4000,
+                "cpu_used_ram": 1000
             }
-            self.report_manager.add_result(
-                report_key='test_report1',
-                result=construct_result(
-                    avg_gpu_metric_values=avg_gpu_metrics,
-                    avg_non_gpu_metric_values=avg_non_gpu_metrics,
-                    comparator=self.result_comparator))
+            self.model_config["name"] = f"test_model1_report_{i}"
+
+            measurement = construct_measurement("test_model1", avg_gpu_metrics,
+                                                avg_non_gpu_metrics,
```

Comment:
Can you refactor the creation of this and the next report out into one function called `_add_report_measurements` or something like that. Let it take the model name and any other important paramters as arguments.

Reply:
Sure